### PR TITLE
Accepting only whole sentences BIN-5 and introduce a testing framework

### PR DESCRIPTION
**Pull request by JonasEtzold**
Contains solution for BIN-5 and BIN-13

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
         logLevel: 3
-        //debug: true
-        //include "log: false" to disable logging
-        //or a "logLevel" integer from 0 to 7 to adjust logging verbosity
-    })
+            //debug: true
+            //include "log: false" to disable logging
+            //or a "logLevel" integer from 0 to 7 to adjust logging verbosity
+    }),
+    sentenceMatcher = new RegExp(`((?:[A-Z]\.|[^\.!?])+)[\.!?]`, "g");
```
Comment (bpoetzschke) [inline index.js:22]: Do you really thing that a sentence has necessarily has to start with a upper case letter? Sometimes when you're lazy sentences also start with a lower case character.

File: index.js [hunk 2]
```diff
@@ -83,38 +93,38 @@ function attachmentGif(w) {
...
+    bot.destroy(bot);
+    persistence.save(words, foundWords, function (err) {
         if (err) {
             throw err;
         } else {
             console.log('Wordlist and found words saved successfully.');
             process.exit(0);
         }
     });
-    if (typeof  bot.destroy === 'function') {
```
Comment (bpoetzschke) [inline index.js:115]: This has to stay in this position here. When `bot.destroy()` is not a method the saving of the words will also fail. And that's not that what we want.

**Review by bpoetzschke — CHANGES_REQUESTED** [review]

File: index.js (hunk 2 above)
Comment (JonasEtzold) [reply index.js:115]: Will fix that

File: index.js (hunk 1 above)
Comment (JonasEtzold) [reply index.js:22]: It will accept also lower case letters in the beginning as this RegEx is not acting case sensitive.

**Review by bpoetzschke — APPROVED** [review]
LGTM

**Pull request merged**